cirq/sim/wave_function.py

```diff
@@ -270,6 +339,23 @@ def _probs(state: np.ndarray, indices: List[int],
     return probs
 
 
+def _single_partial_trace(density_matrix: np.ndarray,
+    n_qubits: int,  index: int) -> np.ndarray:
+    """Helper function to trace out one index from a density matrix."""
+    n_left = index
```

Comment:
I think it's possible to specify this succinctly by using einsum.

I believe the following expression computes the two-qubit density matrix of the second qubit and fifth qubits of a five-qubit state vector v:

```python
v = v.reshape((2,) * 5)
#                 v lower case the included qubits
d = np.einsum('ABCDE,AbCDe->BEbe', v, np.conj(v))
#                         ^ list both cases of included qubits
return d.reshape(4, 4)
```

Reply:
I put this in, in favor of what I had before.